test/e2e/containers_conf_test.go

```diff
@@ -463,6 +463,60 @@ var _ = Describe("Verify podman containers.conf usage", func() {
 		Expect(session.OutputToString()).To(ContainSubstring("test"))
 	})
 
+	Describe("base_hosts_file in containers.conf", func() {
+		var baseHostsFile string
+		var session *PodmanSessionIntegration
+
+		JustBeforeEach(func() {
+			confFile := filepath.Join(podmanTest.TempDir, "containers.conf")
+			err = os.WriteFile(confFile, []byte(fmt.Sprintf("[containers]\nbase_hosts_file=\"%s\"\nno_hosts=false\n", baseHostsFile)), 0755)
+			Expect(err).ToNot(HaveOccurred())
+			os.Setenv("CONTAINERS_CONF_OVERRIDE", confFile)
+			if IsRemote() {
+				podmanTest.RestartRemoteService()
+			}
+
+			session = podmanTest.Podman([]string{"run", "--rm", ALPINE, "cat", "/etc/hosts"})
+			session.WaitWithDefaultTimeout()
+			Expect(session).Should(ExitCleanly())
+		})
+
+		Describe("base_hosts_file=/path/to/hosts", func() {
+			BeforeEach(func() {
+				hostsFile := filepath.Join(podmanTest.TempDir, "hosts")
+				err := os.WriteFile(hostsFile, []byte("12.34.56.78 test.example.com"), 0755)
+				Expect(err).ToNot(HaveOccurred())
+
+				baseHostsFile = hostsFile
+			})
+
+			It("should use the hosts file from the file path", func() {
+				Expect(session.OutputToString()).To(ContainSubstring("12.34.56.78 test.example.com"))
+			})
+		})
+
+		Describe("base_hosts_file=image", func() {
+			BeforeEach(func() {
+				baseHostsFile = "image"
+			})
+
+			It("should use the hosts file from the container image", func() {
+				Expect(session.OutputToString()).To(ContainSubstring("localhost localhost.localdomain"))
+			})
+		})
+
+		Describe("base_hosts_file=none", func() {
+			BeforeEach(func() {
+				baseHostsFile = "none"
+			})
+
+			It("should not use any hosts files", func() {
+				Expect(session.OutputToString()).ToNot(ContainSubstring("localhost.localdomain"))
+				Expect(session.OutputToString()).To(ContainSubstring("localhost"))
```

Comment:
Can you do an exact match for
```
127.0.0.1       localhost
::1     localhost
```
This should be more robust I think.

Also one important bit is that we still add our own custom host entries for the host.containers.internal and the containers name/hostname as well. So I think we need to check that as well for full testing (I know it is not really related to your current change and a testing gap from earlier)
If you run the container with --hostname abc or whatever name you like we could always match that this name is in the file for all cases, i.e. still in the JustBeforeEach() function

Reply:
Thanks added all these